### ISSUE TITLE
[Bug]: KeyError: 'channel'

### ISSUE THREAD

**Issue by Silverdutchman** [tubearchivist/tubearchivist#741]
### I've read the documentation

- [X] I'm running the latest version of Tube Archivist and have read the [release notes](https://github.com/tubearchivist/tubearchivist/releases/latest).
- [X] I have read the [how to open an issue](https://github.com/tubearchivist/tubearchivist/blob/master/CONTRIBUTING.md#how-to-open-an-issue) guide, particularly the [bug report](https://github.com/tubearchivist/tubearchivist/blob/master/CONTRIBUTING.md#bug-report) section.

### Operating System

Unraid

### Your Bug Report

When Rescan subscriptions I get the error - Task extract_download[b2880f8e-ddfe-4021-98f8-812987fa4388] raised unexpected: KeyError('channel')

And I have looked through all the issues (open/closed) but I cannot find a similar error for channels.

I think I am overlooking something, but I cannot see what. So I need your help.

### Relevant log output

```shell
[2024-06-05 11:31:09,591: WARNING/ForkPoolWorker-32] xpcUxwpOQ_A: [32/46]: add to queue
[2024-06-05 11:31:13,901: WARNING/ForkPoolWorker-32] AvDvTnTGjgQ: [33/46]: add to queue
[2024-06-05 11:31:19,539: WARNING/ForkPoolWorker-32] b2880f8e-ddfe-4021-98f8-812987fa4388 Failed callback
[2024-06-05 11:31:19,546: ERROR/ForkPoolWorker-32] Task extract_download[b2880f8e-ddfe-4021-98f8-812987fa4388] raised unexpected: KeyError('channel')
Traceback (most recent call last):
  File "/root/.local/lib/python3.11/site-packages/celery/app/trace.py", line 453, in trace_task
    R = retval = fun(*args, **kwargs)
                 ^^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/celery/app/trace.py", line 736, in __protected_call__
    return self.run(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/home/tasks.py", line 147, in extrac_dl
    videos_added = pending_handler.add_to_pending(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/home/src/download/queue.py", line 254, in add_to_pending
    video_details = self.get_youtube_details(youtube_id, vid_type)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/home/src/download/queue.py", line 327, in get_youtube_details
    return self._parse_youtube_details(vid, vid_type)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/home/src/download/queue.py", line 352, in _parse_youtube_details
    "channel_name": vid["channel"],
                    ~~~^^^^^^^^^^^
KeyError: 'channel'
[2024-06-05 11:31:19,831: WARNING/ForkPoolWorker-32] b2880f8e-ddfe-4021-98f8-812987fa4388 return callback
```

**Comment by bbilly1** [comment]
Interesting, that video does indeed not have a channel: https://www.youtube.com/watch?v=AvDvTnTGjgQ, hence the error. How did you reach that video with a subscription rescan? You should see what channel ID that is part of a little bit above the log snippet you sent. 

But if a video is not part of a channel, yt-dlp wont be able to extract it and TA won't be able to download it. But we can do some error handling, so it can be skipped without stopping the rest of the task.

**Comment by Silverdutchman** [comment]
I think this a real unique case that a video doesn't have a channel attached to it. I didn't realize that until you pointed it out. The video was in one of my oldest playlists. The downside for me was that the scan didn't proceed, for now I have removed the video from the playlist and everything works fine. 

Keep in mind that I am a member of YouTube almost since the beginning and I had to do/undergo several mandatory conversion steps for my account in the past to keep my account. So maybe hence why the video lost it's channel. For example, I have some old video's (hidden) on my own channel that I cannot change anything on it and the only solution YouTube says is to reupload the video's. You can say they are in YouTube Limbo 😁

I thank you for the effort of looking into it. I think this is a very rare occurrence and it should not happen normally.

**Comment by bbilly1** [comment]
fix shipped with v0.4.9, please update.
